spec/ddtrace/contrib/rails/rails_log_auto_injection_spec.rb

```diff
@@ -0,0 +1,151 @@
+require 'ddtrace/contrib/rails/rails_helper'
+
+RSpec.describe 'Rails Rack Rum Injection middleware' do
+  include Rack::Test::Methods
+  include_context 'Rails test application'
+
+  let(:routes) do
+    {
+      '/lograge' => 'lograge_test#index',
+      '/tagged_logging' => 'tagged_logging_test#index'
+    }
+  end
+
+  let(:controllers) do
+    [
+      tagged_logging_controller,
+      lograge_controller
+    ]
+  end
+
+  let(:tagged_logging_controller) do
+    stub_const('TaggedLoggingTestController', Class.new(ActionController::Base) do
+      def index
+        Rails.logger.info('MINASWAN')
+        render inline: '<html> <head> </head> <body> <div> Hello from index </div> </body> </html>'
+      end
+    end)
+  end
+
+  let(:lograge_controller) do
+    stub_const('LogrageTestController', Class.new(ActionController::Base) do
+      def index
+        Rails.logger.info('MINASWAN')
+        render inline: '<html> <head> </head> <body> <div> Hello from index </div> </body> </html>'
+      end
+    end)
+  end
+
+  before do
+    Datadog.configuration[:rails].reset_options!
+    Datadog.configure do |c|
+      c.use :rails, log_injection: true
+    end
+
+    allow(ENV).to receive(:[]).and_call_original
+  end
+
+  after do
+    Datadog.configuration[:rails].reset_options!
+  end
+
+  context 'with Log_Injection Enabled' do
+    # defined in rails support apps
+    let(:logs) { log_output.string }
+
+    if Rails.version >= '3.2'
+      # context 'with Tagged Logging' do
```

Comment:
Are these the tests we talked about offline that are flaky?
Could you clarify what you are seeing if so?

Reply:
When run on their own they appear to pass, but in conjunction with Lograge, or if the Lograge tests are executed first, it appears sometimes the we don't modify the tagged logging logger (causing these tests to fail). My suspicion is we're doing some patching in Lograge tests that we're not unpatching for other tests in the teardown step, I have to look into it more but if you have seen this sort've behavior before in our rails tests i'm open to ideas

Reply:
Try running the same exact test command CI is running, with the same seed and same Ruby version (e.g. `bundle exec appraisal rake spec:rails[--seed=123]`). Most of the time, this allows for reproduction of the issue locally, so you can play around with a debugger/pry.

Reply:
But it very likely could be another leaky test for sure.

Reply:
it was a leaky test, I've been able to resolve this by explicitly disabling `Lograge` when it's related tests are not running, and unsubscribing the `Lograge` `LogSubscriber` in the `reset_rails_configuration` block for `rails3/4/5/6.rb` by calling `Lograge.remove_existing_log_subscriptions`. I've updated these tests to account for this, everything appears to be working as expected now.